sign.go

```diff
@@ -0,0 +1,82 @@
+package floodsub
+
+import (
+	"fmt"
+
+	pb "github.com/libp2p/go-floodsub/pb"
+
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+func verifyMessageSignature(m *pb.Message) error {
+	var pubk crypto.PubKey
+
+	pid, err := peer.IDFromBytes(m.From)
+	if err != nil {
+		return err
+	}
+
+	if m.Key == nil {
+		// no attached key, it must be extractable from the source ID
+		pubk, err = pid.ExtractPublicKey()
+		if err != nil {
+			return err
```

Comment:
maybe log a more specific error here. "could not validate because public key was not available or extractable: $ERR"

Reply:
sure, that's probably a good idea.

Reply:
done